src/lua/compat.lua

```diff
@@ -127,6 +127,16 @@ more verbose representation is used.
 https://tarantool.io/compat/box_error_serialize_verbose
 ]]
 
+local BOX_CONSIDER_SYSTEM_SPACES_SYNCHRONOUS = [[
```

Comment:
Shouldn't compat option switching make all system spaces sync if there's already limbo owner? Current behavior is not good, IMHO:

```
tarantool> box.cfg{}
tarantool> box.ctl.promote()
tarantool> box.info.synchro.queue
---
- owner: 1
  confirm_lag: 0
  term: 2
  age: 0
  len: 0
  busy: false
...
tarantool> require('compat').box_consider_system_spaces_synchronous = 'new'
tarantool> box.space._schema.state
---
- is_sync: false
...

tarantool> box.ctl.promote()
tarantool> box.space._schema.state
---
- is_sync: false
...
tarantool> box.ctl.demote()
tarantool> box.ctl.promote()
tarantool> box.space._schema.state
---
- is_sync: true
...

```

Demote must be done in order to enable force sync. Are we ok with that?

Reply:
We can forbid changing it to 'new', when limbo owner is not nil. As we do with deprecation of `ROLLBACK`: https://github.com/tarantool/tarantool/pull/9869#issuecomment-2208711656

Reply:
@sergepetrenko

Reply:
> I don't think this is a valid use case to support and maintain: the user should set the compat option at the instance start through configuration.

This means the user can only test the new behavior after restarting his service. No good. I think we should allow the user to change the option on the fly in both directions: "old" -> "new" and "new" -> "old". And it should take effect immediately. Not after limbo owner change.

> We can forbid changing it to 'new', when limbo owner is not nil. As we do with deprecation of ROLLBACK: https://github.com/tarantool/tarantool/pull/9869#issuecomment-2208711656

Hmm. Again, no. For example, when raft is enabled, the limbo owner is never nil. It only transitions form one owner to another without demotions. And as I said earlier, we wan't to let the user test the new behavior without restarting the service or performing extra disruptive steps (for example, calling box.ctl.demote() on the current raft leader will cause a leader change and unavailability for rw requests for the time of elections)

Reply:
Thank you for the feedback! 

Added support for transition of the compat option, and added test cases for it.